app/components/member_list/index.js

```diff
@@ -67,18 +72,33 @@ export default class MemberList extends PureComponent {
             rowHasChanged: (r1, r2) => r1 !== r2,
             sectionHeaderHasChanged: (s1, s2) => s1 !== s2
         });
-        const dataSource = props.sections ? ds.cloneWithRowsAndSections(this.createSections(props.members)) : ds.cloneWithRows(props.members);
+        let data = props.members;
+        if (props.showSections) {
+            data = this.createSections(props.members);
+        }
+        const dataSource = ds.cloneWithRowsAndSections(data);
         this.state = {
+            data,
             dataSource
         };
     }
 
     componentWillReceiveProps(nextProps) {
-        const {members, sections} = nextProps;
-        const dataSource = sections ? this.state.dataSource.cloneWithRowsAndSections(this.createSections(members)) : this.state.dataSource.cloneWithRows(members);
-        this.setState({
-            dataSource
-        });
+        const {members, showSections} = nextProps;
+
+        if (members.length !== this.props.members.length || showSections !== this.props.showSections) {
```

Comment:
Would `members !== this.props.members` work here? If any of the profiles change, this won't update, but since you're using a selector, this should be an easy inequality check